libraries/botbuilder-ai/src/localeConverter.ts

```diff
@@ -0,0 +1,212 @@
+/**
+ * @module botbuilder-ai
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Middleware } from 'botbuilder';
+import { Activity, ResourceResponse } from 'botbuilder';
+import * as LanguageMap from './languageMap';
+import * as DateTimeRecoginizers from '@microsoft/recognizers-text-date-time';
+import { Culture } from '@microsoft/recognizers-text-date-time';
+
+/**
+ * The LocaleConverter converts all locales in a message to a given locale.
+ */
+export class LocaleConverterMiddleware implements Middleware {
```

Comment:
>Middleware [](start = 28, length = 10)

We're trying to avoid the use of middleware in class titles.  We went back and forth and currently decided not to use Middleware in titles.